buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -503,7 +503,23 @@ static int writeUtf8(AbstractByteBuf buffer, int writerIndex, CharSequence seq,
 
         // We can use the _set methods as these not need to do any index checks and reference checks.
         // This is possible as we called ensureWritable(...) before.
-        for (int i = 0; i < len; i++) {
+        int i;
+        //ASCII fast-path
+        for (i = 0; i < len; i++) {
+            final char c = seq.charAt(i);
+            if (c < 0x80) {
+                buffer._setByte(writerIndex++, (byte) c);
+            } else {
+                writerIndex = writeUtf8SlowPath(buffer, writerIndex, seq, len, i);
+                break;
+            }
+        }
+        return writerIndex - oldWriterIndex;
+    }
+
+    private static int writeUtf8SlowPath(final AbstractByteBuf buffer, int writerIndex,
+                                         final CharSequence seq, final int len, int i) {
+        for (; i < len; i++) {
             char c = seq.charAt(i);
```

Comment:
Mark c final for consistency?

Reply:
Uhh I've missed it while copying the original body: good point!